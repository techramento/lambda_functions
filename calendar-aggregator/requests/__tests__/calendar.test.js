const calendar = require('../calendar')('APIKEY')
const MockResponses = require('../../__fixtures__/responses.json')
const nock = require('nock')

afterEach(() => nock.cleanAll())

describe('calendar requests', () => {
  it('fetches and processes a specific calendar', () => {
    nock('https://www.googleapis.com')
      .get(/\/calendar\/v3\/calendars\/[^/]+\/events/)
      .reply(200, JSON.stringify(MockResponses.ok))

    // TODO: https://facebook.github.io/jest/docs/expect.html#resolves
    return calendar('SacJS').then((okResponse) =>
      expect(okResponse).toMatchSnapshot())
  })

  it('processes calendar error responses', () => {
    nock('https://www.googleapis.com')
      .get(/\/calendar\/v3\/calendars\/[^/]+\/events/)
      .reply(200, JSON.stringify(MockResponses.error))

    const unexpectedResolve = jest.fn()

    // TODO: https://facebook.github.io/jest/docs/expect.html#rejects
    return calendar('SacJS')
      .then(unexpectedResolve)
      .then(
        (okResponse) => expect(unexpectedResolve).not.toHaveBeenCalled(),
        (errResponse) => expect(errResponse).toMatchSnapshot()
      )
  })
})
