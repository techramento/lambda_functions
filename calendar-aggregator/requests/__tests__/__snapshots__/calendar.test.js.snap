// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calendar requests fetches and processes a specific calendar 1`] = `
Array [
  Object {
    "description": "Name of the Meetup
Thursday, January 01 at 12:00 AM

DESCRIPTION...

https://example.org/",
    "location": "Venue (123 Main Street, Sacramento, CA 95811)",
    "meetup": Object {
      "description": "DESCRIPTION...",
      "group": "Name of the Meetup",
      "short_name": "APIKEY",
      "url": "https://example.org/",
    },
    "start": Object {
      "dateTime": "1970-01-01T00:00:00Z",
    },
    "summary": "Name of the Meetup",
  },
]
`;

exports[`calendar requests processes calendar error responses 1`] = `[TypeError: Cannot read property 'reduce' of undefined]`;
